src/i_video.c

```diff
@@ -527,27 +527,15 @@ static void ProcessEvent(SDL_Event *ev)
 
 static void I_GetEvent(void)
 {
-    #define NUM_PEEP 32
-    static SDL_Event sdlevents[NUM_PEEP];
-
     I_DelayEvent();
 
     SDL_PumpEvents();
 
-    while (true)
```

Comment:
Dammit, I was working on the exact same change just yesterday!

You can keep this code of you increase the array size to `sdlevents[NUM_PEEP + 1]` here, and then pass `sdlevent + 1` as the second parameter to `GetTypedChar()` in `I_HandleKeyboardEvent()`, and use this second parameter as `next_event` in `GetTypedChar()` instead of calling `SDL_PeepEvents()` there.

Reply:
You will have to clear out the value of `next_event.text.text[0]` afterwards, though.

Reply:
As far as I know, we won't need this code in SDL3 because of the improved event handling. So I'm not sure we want to complicate this.